docs/logging.md

```diff
@@ -0,0 +1,160 @@
+`LoggedValue` is a representation of a value that can be written to a log in a string form. 
+It's used internally, and you do not need to use it directly most of the time. Instead, what you want is a `Loggable` 
+typeclass, that describes how a value of some type `A` can be logged, either as a string representation in log message 
+and as a component of structured logging. Given an instance of `Loggable` for a type, a value of the type can be converted 
+into `LoggedValue` and thus logged in a way that you provided. 
+There are multiple predefined ways to create an instance of `Loggable`, many of them can be found in `tofu.logging.Loggable` object:  
+* `Loggable.empty` for no-op logging of value
+* `Loggable.show` for using `cats.Show` instance as string representaion
+* `Loggable.either` for logging either of `A` and `B`
+* provided instances for all primitive types, as well as stdlib's collections and collections from Cats
+* java.time.* instances
+
+Of course, you can describe your `Loggable` instance yourself by extending existing traits that pre-implement some functionality:
+* `DictLoggable` for multi-field objects
+* `ToStringLoggable` for using `.toString` for logging
+* `HideLoggable` to exclude value from logging
+
+#### Loggable example
+```scala mdoc
+import tofu.logging._
+
+case class User(name: String, surname: String)
+
+implicit val userLoggable = new DictLoggable[User] {
+  override def fields[I, V, R, S](a: User, i: I)(implicit r: LogRenderer[I, V, R, S]): R = {
+    r.addString("name", a.name, i)
+    r.addString("surname", a.surname, i)
+  }
+  
+  override def logShow(a: User): String = s"name = ${a.name}, surname = ${a.surname}"
+}
+```
+
+Here, we defined our `Loggable` as a `DictLoggable` that has a notion of `fields`. 
+Fields are a structure of your `Loggable` that is used for structured logging by your `LogRendered`.
+Tofu comes with a `tofu.logging.ElkLayout` (present in `tofu-logging-layout` library), that will log the structure (fields) of your `Loggable`
+as a JSON fields.
+
+#### Syntax extensions
+It's much more convenient to use pre-defined syntax extensions for logging operations since they do all heavy lifting for you:
+```scala mdoc:reset
+import cats.Monad
+import cats.Show
+import cats.syntax.functor._
+import cats.effect._
+
+import tofu.logging._
+import tofu.syntax.logging._
+
+case class User(name: String, surname: String)
+object User {
+  implicit val userShow: Show[User]         = user => s"name=${user.name}, surname=${user.surname}"
+  implicit val userInstance: Loggable[User] = Loggable.show
+}
+
+val user: User = User("name", "surname")
+
+class MyService[F[_]: Logging: Monad] {
+  def doLogging: F[Unit] =
+    for {
+        _ <- info"This is a syntax demo: $user"
+    } yield () 
+}
+
+object MyService {
+  // note, that creating service may require Sync instance, but class itself does not!
+  def apply[F[_]: Sync]: F[MyService[F]] = {
+    val logs = Logs.sync[F, F]
```

Comment:
I would not recommend creating `Logs` per service.
It more like `Logs` per application, so Service initialization would require `Logs[F, F]`

Reply:
Thanks, I made an example a bit clearer with notes